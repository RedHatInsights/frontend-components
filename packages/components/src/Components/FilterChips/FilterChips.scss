.ins-c-chip-group__plain {

}
