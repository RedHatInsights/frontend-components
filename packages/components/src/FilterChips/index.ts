export { default } from './FilterChips';
export { default as FilterChips } from './FilterChips';
export * from './FilterChips';
