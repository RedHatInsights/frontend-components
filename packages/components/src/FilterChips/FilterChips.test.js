import React from 'react';
import FilterChips from './FilterChips';
import { render } from '@testing-library/react';

const filters = [
  {
    category: 'Group 1',
    chips: [
      {
        name: 'Chip 1',
        isRead: true,
      },
      {
        name: 'Chip 2',
        count: 3,
      },
    ],
  },
  {
    name: 'Chip 3',
  },
  {
    name: 'Chip 4',
  },
];

describe('FilterChips component', () => {
  it('should render - no data', () => {
    const { container } = render(<FilterChips />);
    expect(container).toMatchSnapshot();
  });

  it('should render', () => {
    const { container } = render(<FilterChips filters={filters} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with custom delete message', () => {
    const { container } = render(<FilterChips filters={filters} deleteTitle={<div>Some delete title</div>} />);
    expect(container).toMatchSnapshot();
  });

  it('should render group delete', () => {
    const { container } = render(<FilterChips filters={filters} onDeleteGroup={() => undefined} />);
    expect(container).toMatchSnapshot();
  });

  describe('API', () => {
    it('should call onDelete when deleting a single chip', () => {
      const onDelete = jest.fn();
      const { container } = render(<FilterChips filters={filters} onDelete={onDelete} />);
      const chips = container.querySelectorAll('.pf-v6-c-label button');
      chips[chips.length - 1].click();
      expect(onDelete).toHaveBeenCalledWith(expect.anything(), [{ name: 'Chip 4' }]);
      expect(onDelete).toHaveBeenCalledTimes(1);
    });

    it('should call onDelete when deleting a single chip in group', () => {
      const onDelete = jest.fn();
      const { container } = render(<FilterChips filters={filters} onDelete={onDelete} />);
      container.querySelectorAll('.pf-v6-c-label button')[0].click();
      expect(onDelete).toHaveBeenCalledWith(expect.anything(), [
        {
          category: 'Group 1',
          chips: [
            {
              name: 'Chip 1',
              isRead: true,
            },
          ],
        },
      ]);
      expect(onDelete).toHaveBeenCalledTimes(1);
    });

    it('should call onDelete when deleting all chips', () => {
      const onDelete = jest.fn();
      const { container } = render(<FilterChips filters={filters} onDelete={onDelete} />);
      const buttons = container.querySelectorAll('button');
      buttons[buttons.length - 1].click();
      expect(onDelete).toHaveBeenCalledWith(expect.anything(), filters, true);
      expect(onDelete).toHaveBeenCalledTimes(1);
    });

    it('should not call onDelete when clicking on any ChipGroup', () => {
      const onDelete = jest.fn();
      const { container } = render(<FilterChips filters={filters} onDelete={onDelete} />);

      container.querySelectorAll('.pf-v6-c-label-group').forEach((group) => group.click());
      expect(onDelete).not.toHaveBeenCalled();
    });

    it('should call onDelete when deleting a single chip', () => {
      const onDelete = jest.fn();
      const newGroup = {
        category: 'Group 2',
        chips: [
          {
            name: 'Chip 1',
          },
          {
            name: 'Chip 2',
          },
        ],
      };
      const { container } = render(<FilterChips filters={[...filters, newGroup]} onDeleteGroup={onDelete} />);
      const buttons = container.querySelectorAll('.pf-v6-c-label-group__close button');
      buttons[buttons.length - 1].click();
      expect(onDelete).toHaveBeenCalledWith(
        expect.anything(),
        [newGroup],
        [
          {
            category: 'Group 1',
            chips: [
              {
                name: 'Chip 1',
                isRead: true,
              },
              {
                name: 'Chip 2',
                count: 3,
              },
            ],
          },
        ],
      );
      expect(onDelete).toHaveBeenCalledTimes(1);
    });
  });
});
