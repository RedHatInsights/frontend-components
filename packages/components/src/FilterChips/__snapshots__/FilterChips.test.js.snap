// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterChips component should render - no data 1`] = `
<div>
  <span
    class="ins-c-chip-filters"
  />
</div>
`;

exports[`FilterChips component should render 1`] = `
<div>
  <span
    class="ins-c-chip-filters"
  >
    <div
      class="pf-v6-c-label-group pf-m-category"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <span
          aria-hidden="true"
          class="pf-v6-c-label-group__label"
          id="pf-random-id-0"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-0"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-1"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 1
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 1,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-2"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 2
                  <span
                    class="pf-v6-c-badge pf-m-unread"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 2,[object Object]"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-2"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-3"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 3
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 3,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-3"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-4"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 4
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 4,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-4"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v6-c-button pf-m-link"
      data-ouia-component-id="ClearFilters"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-button__text"
      >
        Clear filters
      </span>
    </button>
  </span>
</div>
`;

exports[`FilterChips component should render group delete 1`] = `
<div>
  <span
    class="ins-c-chip-filters"
  >
    <div
      class="pf-v6-c-label-group pf-m-category"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <span
          aria-hidden="true"
          class="pf-v6-c-label-group__label"
          id="pf-random-id-6"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-6"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-9"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 1
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 1,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-9"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-10"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 2
                  <span
                    class="pf-v6-c-badge pf-m-unread"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 2,[object Object]"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-10"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
      <div
        class="pf-v6-c-label-group__close"
      >
        <button
          aria-disabled="false"
          aria-label="Close chip group"
          aria-labelledby="remove_group_pf-random-id-6 pf-random-id-6"
          class="pf-v6-c-button pf-m-plain pf-m-no-padding"
          data-ouia-component-id="OUIA-Generated-Button-plain-11"
          data-ouia-component-type="PF6/Button"
          data-ouia-safe="true"
          id="remove_group_pf-random-id-6"
          type="button"
        >
          <span
            class="pf-v6-c-button__icon"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-11"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 3
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 3,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-12"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-12"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 4
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 4,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-13"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v6-c-button pf-m-link"
      data-ouia-component-id="ClearFilters"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-button__text"
      >
        Clear filters
      </span>
    </button>
  </span>
</div>
`;

exports[`FilterChips component should render with custom delete message 1`] = `
<div>
  <span
    class="ins-c-chip-filters"
  >
    <div
      class="pf-v6-c-label-group pf-m-category"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <span
          aria-hidden="true"
          class="pf-v6-c-label-group__label"
          id="pf-random-id-3"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-3"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-5"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 1
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 1,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-5"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-6"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 2
                  <span
                    class="pf-v6-c-badge pf-m-unread"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 2,[object Object]"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-6"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-7"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 3
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 3,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-7"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="pf-v6-c-label-group"
    >
      <div
        class="pf-v6-c-label-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v6-c-label-group__list"
          data-ouia-component-type="PF6/ChipGroup"
          data-ouia-safe="true"
          role="list"
        >
          <li
            class="pf-v6-c-label-group__list-item"
          >
            <span
              class="pf-v6-c-label pf-m-outline"
              data-ouia-component-id="OUIA-Generated-Chip-8"
              data-ouia-component-type="PF6/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v6-c-label__content"
              >
                <span
                  class="pf-v6-c-label__text"
                  style="--pf-v6-c-label__text--MaxWidth: 16ch;"
                >
                  Chip 4
                </span>
              </span>
              <span
                class="pf-v6-c-label__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Chip 4,"
                  class="pf-v6-c-button pf-m-plain pf-m-no-padding"
                  data-ouia-component-id="OUIA-Generated-Button-plain-8"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-v6-c-button__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v6-c-button pf-m-link"
      data-ouia-component-id="ClearFilters"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-button__text"
      >
        <div>
          Some delete title
        </div>
      </span>
    </button>
  </span>
</div>
`;
